R/disambiguate.R

```diff
@@ -56,37 +56,51 @@ get_table_colnames <- function(dm, tables = NULL) {
     anti_join(keep_colnames, by = c("table", "column"))
 }
 
+#' create a disambiguation recipe tibble
+#'
+#' It will contain :
+#'   * table: the table name
+#'   * renames: a list of named symbols to be substituted in
+#'     `db_rename(dm, tbl, new = old)`
+#'   * name and a list of tibbles containing character cols `new_name` and `column`
+#'     that will be used to print`db_rename` instructions through explain_col_rename
+#' @param table_colnames a table containing table name and col names of dm
+#' @param sep separator used to create new names for dupe cols
+#' @noRd
 compute_disambiguate_cols_recipe <- function(table_colnames, sep) {
   dupes <- vec_duplicate_detect(table_colnames$column)
   dup_colnames <- table_colnames[dupes, ]
 
   dup_colnames$new_name <- paste0(dup_colnames$table, sep, dup_colnames$column)
   dup_data <- dup_colnames[c("new_name", "column")]
-  dup_data$column <- syms(dup_data$column)
+  dup_data$column_sym <- syms(dup_data$column)
 
   dup_nested <-
     vec_split(dup_data, dup_colnames$table) %>%
     set_names("table", "renames")
 
-  dup_nested$renames <- map(dup_nested$renames, deframe)
+  dup_nested$names <- map(dup_nested$renames, select, new_name, column)
+  dup_nested$renames <- map(dup_nested$renames, ~deframe(select(., -column)))
   as_tibble(dup_nested)
 }
 
+
+#' Describe renaming of cols by printing code
+#'
+#' @param recipe created by `compute_disambiguate_cols_recipe`
+#' @noRd
 explain_col_rename <- function(recipe) {
   if (nrow(recipe) == 0) {
     return()
   }
 
-  msg_base <-
+  disambiguation <-
     recipe %>%
-    mutate(renames = map(renames, ~ enframe(., "new", "old"))) %>%
-    unnest_df("renames", tibble(new = character(), old = syms(character()))) %>%
-    nest(data = -old)
-
-  sub_text <- map_chr(msg_base$data, ~ paste0(.x$new, collapse = ", "))
-  msg_core <- paste0("* ", msg_base$old, " -> ", sub_text)
+    unnest(names) %>%
+    mutate(text = glue("dm_rename({tick_if_needed(table)}, {tick_if_needed(new_name)} = {tick_if_needed(column)})")) %>%
+    pull(text)
 
-  message("Renamed columns:\n", paste(msg_core, collapse = "\n"))
+  message("Renaming ambiguous columns: %>%\n  ", glue_collapse(disambiguation, " %>%\n  "))
```

Comment:
Should we extract a function from here so that we use the same pattern for all instances where we print code?

Maybe in a separate PR.